app/src/main/java/com/github/se/cyrcle/io/PreferenceStorage.kt

```diff
@@ -0,0 +1,41 @@
+package com.github.se.cyrcle.io
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.preferencesDataStore
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+
+val Context.preferencesDataStore: DataStore<Preferences> by preferencesDataStore(name = "settings")
+
+/** A class to simplify reading and writing preferences to the preference data store. */
+class PreferenceStorage(private val context: Context) {
+
+  /**
+   * Write a preference associated with [preferenceKey] to the data store. If the value is already
+   * present, it will be overwritten.
+   *
+   * @param preferenceKey The key of the preference to write.
+   * @param value The value to associate with [preferenceKey].
+   */
+  suspend fun <T> writePreference(preferenceKey: Preferences.Key<T>, value: T) {
```

Comment:
Sadly, there is no easy way to ensure that no one overwrite anything. I thought of having a list right above it, but I already know it won't be up-to-date with everybody.
I also thought of having an Enum to solve this, but at this point, it was more obfuscation than it would be helping anybody.
I guess it will be up to the team not to mess this.

Reply:
I see, and understand that these things are sometimes unavoidable. We'll have to be careful when implementing offline mode